lib/database/migrations/20240201102034-create-quality-control-flags-tables.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+        await queryInterface.createTable('quality_control_flag_reasons', {
+            // Properties
+            id: {
+                type: Sequelize.INTEGER,
+                allowNull: false,
+                primaryKey: true,
+                autoIncrement: true,
+            },
+            name: {
+                type: Sequelize.STRING,
+                allowNull: false,
+                unique: true,
+            },
+            method: {
+                type: Sequelize.STRING,
+                unique: true,
+                allowNull: false,
+            },
+            bad: {
+                type: Sequelize.BOOLEAN,
+                allowNull: false,
+            },
+            archived_at: {
+                type: Sequelize.DATE,
+                allowNull: true,
+            },
+
+            // Timestamps
+            created_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP'),
+            },
+            updated_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'),
+            },
+        }, { transaction });
+
+        await queryInterface.createTable('quality_control_flags', {
+            // Properties
+            id: {
+                type: Sequelize.INTEGER,
+                allowNull: false,
+                primaryKey: true,
+                autoIncrement: true,
+            },
+            from: {
+                type: Sequelize.DATE,
+                allowNull: false,
+            },
+            to: {
+                type: Sequelize.DATE,
+                allowNull: false,
+            },
+            comment: {
+                type: Sequelize.TEXT,
+            },
+            provenance: {
+                type: Sequelize.ENUM('HUMAN', 'SYNC', 'ASYNC', 'MC'),
+                allowNull: false,
+            },
```

Comment:
Why do you want to add `"HUMAN"` here?
For me, there is a model issue here, and I think that this property is not useful.
The `data provenance` info is already stored in the related pass.
I think that you should have two different flags (or at least, flag links): one for apass and cpass (links to `data_pass`) and one for mc pass (links to `simulation_pass`).
To avoid duplication, if flags are actually the same entity, I would keep this table, but remove the `data_pass_id` from here, and create two relation table: 
- `data_pass_quality_control_flag` linking `quality_control_flag` to `data_pass`
- `simulation_pass_quality_control_flag` linking `quality_control_flag` to `simulation_pass`

Because with the current model, you can't link your flags to simulation passes.

Reply:
ok, I neglected that qc Flag might defined for MC, all applied